### PR TITLE
Allow -v/--verbose anywhere in command line

### LINKED ISSUE #711: Allow --verbose flag to be passed in sub-commands
**Description**

When using `python -m sigstore verify identity ...` I received a recommendation from the resulting error to use `--verbose` for more information.

Providing the `--verbose` flag along with other CLI flags resulted in a confusing error message about the `--verbose` flag not being supported:

```
$ python -m sigstore verify identity --verbose ...
```

But providing the `--verbose` flag to the `sigstore` namespace before the `verify identity` subcommands worked as expected:

```
$ python -m sigstore --verbose verify identity ...
```

Most sub-command CLIs let you pass arguments to the sub-command rather than needing to provide them to an earlier namespace.

**Version**

```
$ python -m sigstore --version
sigstore 1.1.2
```

### PR DESCRIPTION
**Pull request by jku**
#### Summary

Allow "-v/--verbose" after commands and subcommands:
```
  sigstore -v sign FILE
  sigstore sign -v FILE

```

This was slightly non-trivial:
* the parent parser approach works, but every parser sets the default value, meaning in practice only the last handled parsers value is used
* argparse.SUPPRESS as a default value makes this work so that parsers only set a value if the option was used
* Suppressing then requires handling the no-verbose case after parsing

Multiple "-v" are still supported but not in multiple different commands (so "sigstore -v sign -v FILE" results in verbose=1).

This fixes #711

#### Release Note

NONE: I don't think this requires a release note mention

#### Documentation

 The "--help" output slightly changes with this commit (the subcommands list verbose as option).
README is updated and passes `make check-readme`

### CODE REVIEW

**Comment by jku** [comment]
I'm unsure where the `make check-readme` failure comes from:
```
< options:
> optional arguments:
```

I switched to "options" because that's what make check-readme tells me to do locally :shrug: I will change it back

**Comment by woodruffw** [comment]
> I switched to "options" because that's what make check-readme tells me to do locally 🤷 I will change it back

Yeah, this is annoying: it's because Python 3.10 or newer (IIRC) changes the default `argparse` help rendering. That's why we do the check with 3.9 in CI. Thanks for reverting, sorry for the friction there 🙂

File: sigstore/_cli.py [hunk 1]
```diff
@@ -235,20 +235,24 @@ def _add_shared_oidc_options(
...
     parser.add_argument(
-        "-V", "--version", action="version", version=f"%(prog)s {__version__}"
-    )
-    parser.add_argument(
-        "-v",
-        "--verbose",
-        action="count",
-        default=0,
-        help="run with additional debug logging; supply multiple times to increase verbosity",
+        "-V", "--version", action="version", version=f"sigstore {__version__}"
```
[the remark this answers is not in the document]
Comment (woodruffw) [reply sigstore/_cli.py:255]: Yeah, good call.

**Review by woodruffw — APPROVED** [review]
LGTM, thanks a ton @jku! I agree about not needing a changelog entry.

@sethmlarson would you mind giving this changeset a test and seeing if it fills your use case?

**Comment by sethmlarson** [comment]
I installed this PR locally and everything appears to be working as expected, thanks @jku!

**Comment by woodruffw** [comment]
Fantastic! I merged with `main` here, so I've pinged @di to do another review to pass our approval requirements.

**Review by di — APPROVED** [review]

**Pull request merged**